mautrix_signal/portal.py

```diff
@@ -1691,8 +1692,20 @@ async def _get_power_levels(
                 ac = info.access_control
                 for detail in info.member_detail + info.pending_member_detail:
                     puppet = await p.Puppet.get_by_address(Address(uuid=detail.uuid))
-                    level = 50 if detail.role == GroupMemberRole.ADMINISTRATOR else 0
-                    levels.users[puppet.intent_for(self).mxid] = level
+                    puppet_mxid = puppet.intent_for(self).mxid
+                    current_level = levels.get_user_level(puppet_mxid)
+                    if bot_pl > current_level and bot_pl >= 50:
+                        level = current_level
+                        if await u.User.get_by_address(Address(uuid=detail.uuid)):
```

Comment:
Seems like it would be easier to just check if the mxid is a double puppet:

```suggestion
                        if puppet.is_real_user:
```